### PR TITLE
Feat edit delete post

### PR DESCRIPTION
**Pull request by SinchanaMS**
This request adds the following features:

1. Edit Post
2. Delete Post
3. Code refactor - Segregated Firebase calls

### Preview:
1. Post before edit:
image

2. Post after edit: 
image

### CODE REVIEW

**Review by Adarsh-Patel-Dev — APPROVED** [review]
No changes are required in the code
I like how you kept firebaseCalls & firebaseListeners in different files
It would be better if you've provided a live link in PR

**Review by rutvikpumak — APPROVED** [review]
Code looks good and well structured.
Functionality working as expected

**Comment by SinchanaMS** [comment]
Thank you for the reviews! Merging to dev.

File: src/features/user/Signup.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default function Signup() {
     name: "",
     email: "",
     password: "",
-    profilePic: "",
+    profilePic: "http://cdn.onlinewebfonts.com/svg/img_264570.png",
```
Comment (keyurshah339) [inline src/features/user/Signup.jsx:18]: put the image in the codebase and use never use a direct link

**Pull request merged**